cmd/image-converter/image-converter.go

```diff
@@ -57,21 +85,20 @@ func convert(srcVolPath, dstVolPath, sourceFormat, targetFormat string) error {
 		return err
 	}
 
-	// Replace /mnt/disk.img with /output/disk.img
-	cmd = exec.Command(
-		"mv",
-		dstVolPath,
-		srcVolPath,
-	)
+	return nil
+}
 
-	if err := cmd.Start(); err != nil {
-		return err
+func isBlockDevice(path string) bool {
+	fileInfo, err := os.Stat(path)
+	if err != nil {
+		klog.Fatal(err)
 	}
 
-	err = cmd.Wait()
-	if err != nil {
-		return err
+	stat, ok := fileInfo.Sys().(*syscall.Stat_t)
+	if !ok {
+		klog.Fatal("failed to get raw syscall.Stat_t data")
 	}
 
-	return nil
+	// Check if it's a block device
+	return (stat.Mode & syscall.S_IFMT) == syscall.S_IFBLK
```

Comment:
what happens on rbd that supports both (when we get a filesystem on block storage)?

Reply:
it depends on the volumeMode on the PVC, which determines how it's attached to the node, so it shouldn't matter

Reply:
ok, then (stat.Mode & syscall.S_IFMT) will return a filesystem in that case and both source and destination volumes are of the same mode so it looks correct, but why not to propagate this from the controller that knows the mode already?

Reply:
i think it's weird from the standpoint of the container to request something that can be determined

Reply:
yeah, it's less standalone but on the other hand it's simplify the code - I think it looks better now, thanks for changing that